src/inode2prog.cpp

```diff
@@ -151,9 +142,21 @@ void setnode(unsigned long inode, pid_t pid) {
     prg_node *newnode = new prg_node;
     newnode->inode = inode;
     newnode->pid = pid;
-    newnode->cmdline = getcmdline(pid);
+    bool result = getcmdline(pid, newnode->cmdline);
```

Comment:
Perhaps we should split this into 2 parameters (one for the program and one for the parameters) already here?

Reply:
Sorry for the delay.
If the caller needs all the details, `newnode->cmdline` need to be of type `vector<string>` , because there may be spaces in an parameter.

Reply:
If I'm reading this right, `newnode->cmdline` is now populated with a string containing the program name, then a `\0`, and then all parameters (with spaces).

That seems kind of odd. I was proposing returning 2 strings instead, one for the program name and one for the parameters (with spaces), removing the funny business with the `\0` character. WDYT?